src/ChipCarousel/ChipCarousel.test.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import ChipCarousel from '.';
+
+const items = [
+  {
+    props: {
+      onClick: jest.fn(),
+      children: 'Item 1',
+    },
+  },
+  {
+    props: {
+      onClick: jest.fn(),
+      children: 'Item 2',
+    },
+  },
+  {
+    props: {
+      onClick: jest.fn(),
+      children: 'Item 3',
+    },
+  },
+  {
+    props: {
+      onClick: jest.fn(),
+    },
+  },
+  'Test string',
+];
+
+const mockScrollTo = jest.fn();
+Element.prototype.scrollTo = mockScrollTo;
+
+const ariaLabel = 'Test aria label';
+function TestingChipCarousel(props) {
+  return (
+    <IntlProvider locale="en">
+      <ChipCarousel ariaLabel={ariaLabel} items={items} {...props} />
+    </IntlProvider>
+  );
+}
+
+describe('<ChipCarousel />', () => {
+  it('should render the carousel correctly', () => {
+    render(<TestingChipCarousel />);
+
+    const carousel = document.getElementsByClassName('pgn__chip-carousel');
```

Comment:
Is there a way around needing to rely on the implementation details of the component in these tests by relying more on the `@testing-library/react` (RTL) API?

This is kind of using RTL almost as if it were Enzyme by relying on implementation details like class names. Perhaps we rely more on `data-testid` attributes here for selections?

```jsx
function TestingChipCarousel(props) {
  return (
    <IntlProvider locale="en">
      <ChipCarousel data-testid="chip-carousel" ariaLabel={ariaLabel} items={items} {...props} />
    </IntlProvider>
  );
}

render(<TestingChipCarousel />);

const carousel = screen.getByTestId('chip-carousel');
```

Similarly for the `chipItems` below, we could do something like add a `data-testid="chip-component"` on each child object passed to `ChipCarousel` such that we could query using RTL like: `screen.queryByTestId('chip-component')`

Reference: https://kentcdodds.com/blog/testing-implementation-details

Reply:
Agree on this point. Refactored tests to use `data-testid`